controller/source/rocksdb_server/message.hpp

```diff
@@ -0,0 +1,152 @@
+    static std::unordered_set<std::string> valid_query_types {"get", "put", "del"};
+
+    std::vector<std::string> splits;
+    boost::split(splits, raw_query, boost::is_any_of(" "));
+
+    if (splits.size() < 2) {
+      std::cout << "Invalid query message. Minimum 2 arguments needed."
+                << std::endl;
+      return invalid_query;
+    }
+
+    if (!valid_query_types.contains(splits[0])) {
+      std::cout
+          << "Invalid query type. A valid query can be one of {get, put, del}"
+          << std::endl;
+      return invalid_query;
+    }
+
+    if (splits[0] == "put" && splits.size() != 3) {
+      std::cout << "Invalid query. put query expects exactly 3 arguments."
+                << std::endl;
+      return invalid_query;
+    }
+
+    query_message result;
+    result.m_command = splits[0];
+    result.m_key = splits[1];
+    if (splits.size() == 3) {
+      result.m_value = splits[2];
+    }
+    result.m_is_valid = true;
+    return result;
+  }
+
+  auto get_command() -> std::string {
+    return m_command;
+  }
+
+  void set_command(const std::string& command) {
+    m_command = command;
+  }
+
+  auto get_key() -> std::string {
+    return m_key;
+  }
+
+  void set_key(const std::string& key) {
+    m_key = key;
+  }
+
+  auto get_value() -> std::string {
+    return m_value;
+  }
+
+  void set_value(const std::string& value) {
+    m_value = value;
+  }
+
+  auto get_is_valid() const -> bool {
+    return m_is_valid;
+  }
+
+  void set_is_valid(bool is_valid) {
+    m_is_valid = is_valid;
+  }
+  
+private:
+  std::string m_command;
+  std::string m_key;
+  std::string m_value;
+  bool m_is_valid {false};
+};
+
+class response_message
+{
+public:
+  response_message() = default;
+
+  response_message(bool is_success, std::string data)
+      : m_is_success {is_success}
+      , m_data {std::move(data)}
+  {
+  }
+
+  auto serialize() -> std::string
+  {
+    std::string result;
+    result.append(m_is_success ? "success: " : "failure: ")
+        .append(m_data)
+        .append("\n");
+    return result;
+  }
+
+  static auto deserialize(const std::string& raw_query) -> response_message
+  {
+    static response_message invalid_response {/*is_success=*/false, "invalid"};
+    static std::unordered_set<std::string> valid_response_types {"success", "failure"};
+
+    size_t index = raw_query.find(": ");
+    if (index == std::string::npos) {
+      return invalid_response;
+    }
+
+    std::string first_str = raw_query.substr(0, index);
+    if (!valid_response_types.contains(first_str)) {
+      return invalid_response;
+    }
+
+    std::string second_str = raw_query.substr(index + 2, raw_query.size());
+    return response_message {first_str == "success", second_str};
+  }
+
+  auto get_is_success() const -> bool {
+    return m_is_success;
+  }
+
+  auto get_data() -> std::string {
+    return m_data;
+  }
+
+private:
+  bool m_is_success {false};
+  std::string m_data;
```

Comment:
Maybe add a comment indicating the current message format for get/put/delete with an example?
That would be helpful for future adaptations. @ertugrulaypek

Reply:
added docstring with examples